pyiron_base/state/settings.py

```diff
@@ -372,10 +379,48 @@ def _get_config_from_environment(self) -> Union[Dict, None]:
             elif k in self.environment_credential_map:
                 config[self.environment_credential_map[k]] = v
         config = self._fix_boolean_var_in_config(config=config)
-        config = self._add_credentials_from_file(config)
         return config if len(config) > 0 else None
 
-    def _add_credentials_from_file(self, config: dict) -> Dict:
+    def _add_credentials_from_file(self) -> Dict:
+        if (
+            "credentials_file" in self._configuration
+            and self._configuration["credentials_file"] is not None
+        ):
+            credential_file = self._configuration["credentials_file"]
+
+            # This gets all the entries in the credential file with the headers
+            # The key-value pairs in the [DEFAULT] section are added everywhere!
+            parser = ConfigParser(inline_comment_prefixes=(";",), interpolation=None)
+            parser.read(credential_file)
+            credentials = {}
+            for sec_name, section in parser.items():
+                credentials_w = {}
+
+                for k, v in section.items():
+                    if k.upper() in self.file_credential_map:
+                        credentials_w[self.file_configuration_map[k.upper()]] = v
+                    else:
+                        credentials_w[k.lower()] = v
+                if len(credentials_w) > 0:
+                    credentials[sec_name.upper()] = credentials_w
+            return credentials
+
+    def _update_credentials_from_std_pyiron_config(self):
+        pyiron_dict_name = "PYIRON"
```

Comment:
I'm not diving deep into the surrounding scope here, but it smells a little bit like this should be a (module-level) global `PYIRON_DICT_NAME = "PYIRON"`. You will have a much better sense whether or not that's the case, I just want to nudge the consideration.